pybamm/models/submodels/external_circuit/base_external_circuit.py

```diff
@@ -41,23 +42,25 @@ def get_fundamental_variables(self):
         return variables
 
     def set_initial_conditions(self, variables):
-        Q_Ah = variables["Discharge capacity [A.h]"]
-        Qt_Ah = variables["Throughput capacity [A.h]"]
-        self.initial_conditions[Q_Ah] = pybamm.Scalar(0)
-        self.initial_conditions[Qt_Ah] = pybamm.Scalar(0)
+        if self.add_discharge_capacity:
```

Comment:
We add these by default when using this as a submodel, but pass `add_discharge_capacity=False` when creating the submodel to update the "Current [A]" variable for experiments to avoid adding these equations to models that don't already contain them. I'd welcome alternative implementations.

Reply:
How about just having a separate submodel for discharge capacity and related variables?